### PR TITLE
Authorization

### PR DESCRIPTION
**Pull request by dkupchyk**
Implemented sign in and sign up forms and made then automatically render with eventemitter.

### CODE REVIEW

File: src/scripts/components/Auth.js [hunk 1]
```diff
@@ -1,19 +1,95 @@
...
+
+  signInFormSubmit(e) {
+    e.preventDefault();
+
+    AuthService.Login({
+      identifier: this.signinUsername.value,
+      password: this.signinPassword.value,
+    });
+
+    emitter.subscribe('authorized', this.render);
```
Comment (Ikilonchic) [inline src/scripts/components/Auth.js]: You can subscribe to the event only once in the registerListener() method.

Comment (dkupchyk) [reply src/scripts/components/Auth.js]: What about subscribing to event listener in App once?

File: src/scripts/services/AuthService.js [hunk 2]
```diff
@@ -15,6 +16,7 @@ export default class AuthService {
     });
 
     User.token = response && response.jwt;
+    emitter.emit('authorized');
```
Comment (Ikilonchic) [inline src/scripts/services/AuthService.js]: The user is only authorized if the response is not null. Move emit under the condition.

File: src/scripts/services/AuthService.js [hunk 3]
```diff
@@ -29,5 +31,6 @@ export default class AuthService {
...
```
Comment (Ikilonchic) [inline src/scripts/services/AuthService.js]: Same.

File: src/scripts/components/Auth.js (hunk 1 above)
Comment (Ikilonchic) [reply src/scripts/components/Auth.js]: Fine, then we can add the Dashboard render to the callback.

Comment (dkupchyk) [reply src/scripts/components/Auth.js]: Done. Moved to App file.

File: src/scripts/services/AuthService.js (hunk 2 above)
Comment (dkupchyk) [reply src/scripts/services/AuthService.js]: Done.

File: src/scripts/services/AuthService.js (hunk 3 above)
Comment (dkupchyk) [reply src/scripts/services/AuthService.js]: Done.

**Comment by Ikilonchic** [comment]
Everything is fine.

**Pull request merged**